CMakeLists.txt

```diff
@@ -29,9 +29,11 @@ set(ASIO_TAG
 
 option(ASIO_USE_CPM "Download Asio with CPM instead of FetchContent" OFF)
 option(ASIO_CPM_FETCHCONTENT_COMPAT "Should asio be declared with FetchContent functions to be compatible. This doesn't not allow CPM cache to work." ON)
```

Comment:
@OlivierLDff Q: why do use still use FetchContent?

Reply:
To keep compatibility with some old projects, but right, I should update them anyway. 
I will split your PR in multiple smaller PR